src/core/socialprovider.js

```diff
@@ -363,8 +365,14 @@ XMPPSocialProvider.prototype.onMessage = function(msg) {
  * @param {String} msgs A batch of messages.
  */
 XMPPSocialProvider.prototype.receiveMessage = function(from, msgs) {
-  var i, parsedMessages = JSON.parse(msgs);
-  for (i = 0; i < parsedMessages.length; i+=1) {
+  var parsedMessages;
+  try {
```

Comment:
To be safe, we should make sure the parsed Object is in the format that you want.
For example:
JSON.parse("{key: value}") = {key: value}
JSON.parse("[1,2,3]") = [1,2,3]
JSON.parse("true") = true
JSON.parse("1") = 1
etc etc etc.